### PR TITLE
Use `typed_pos_args` in fs module

### PR DESCRIPTION
**Pull request by dcbaker**
This is just a single commit from my larger series to use `typed_pos_args` in all of meson, plus a couple of easy optimizations I saw.

### CODE REVIEW

File: mesonbuild/modules/fs.py [hunk 1]
```diff
@@ -60,69 +59,57 @@ def _resolve_dir(self, state: 'ModuleState', arg: str) -> Path:
...
-    @stringArgs
     @noKwargs
     @FeatureNew('fs.expanduser', '0.54.0')
-    def expanduser(self, state: 'ModuleState', args: T.Sequence[str], kwargs: dict) -> ModuleReturnValue:
-        if len(args) != 1:
-            raise MesonException('fs.expanduser takes exactly one argument.')
-        return ModuleReturnValue(str(Path(args[0]).expanduser()), [])
+    @typed_pos_args('fs.expanduser', str)
+    def expanduser(self, state: 'ModuleState', args: T.Tuple[str], kwargs: T.Dict[str, T.Any]) -> ModuleReturnValue:
+        return ModuleReturnValue(os.path.expanduser(args[0]), [])
```
Comment (xclaesse) [inline mesonbuild/modules/fs.py:72]: I thought we wanted to use pathlib everywhere?

Comment (xclaesse) [reply mesonbuild/modules/fs.py:72]: Ok forget me, you explain in the commit message. I don't think performance matter in this case, but it's indeed not worth doing conversion from/to str.

**Comment by xclaesse** [comment]
Looks good to me, but CI is red.

**Comment by dcbaker** [comment]
Yup, I think it's that cleanup pass. Apparently `Path.something` is different than `os.path.something`.

**Comment by xclaesse** [comment]
> Yup, I think it's that cleanup patch. Apparently `Path.something` is different than `os.path.something`.

Yep, it's full of subtle differences.

**Pull request merged**